ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMiningCoordinator.java

```diff
@@ -152,7 +147,7 @@ private void startAsyncMiningOperation() {
         executor.startAsyncMining(minedBlockObservers, ethHashObservers, parentHeader);
   }
 
-  private synchronized boolean haltCurrentMiningOperation() {
+  protected synchronized boolean haltCurrentMiningOperation() {
```

Comment:
nit: changing this from private to protected opens the API significantly (and just for logging purposes) - is there a way we can move the conditional into the child-class, but keep the heavy-lifting here?

Reply:
The new implementation no longer needs this

Reply:
I created two methods `onPauseMining` and `onResumeMining` which allow to keep the main implementation in the abstract class